yagmail/message.py

```diff
@@ -212,10 +214,8 @@ def get_mime_object(is_marked_up, content_string, encoding):
         return content_object
 
     if content_object["main_type"] is None:
-        if isinstance(content_string, str):  # Only guess type if it is an URL (i.e. a string)
-            content_type, _ = mimetypes.guess_type(content_string)
-        else:
-            content_type = None
+        # Guess the mimetype with the filename
+        content_type, _ = mimetypes.guess_type(content_name)
```

Comment:
Are you sure this is correct and cannot lead to an error?

Reply:
the `content_name` is always a string and therefore accepted by the `guess_type` function. IMO worst case scenario would be that `(None, None)` is returned by `guess_type`